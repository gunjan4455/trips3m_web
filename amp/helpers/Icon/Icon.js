import React from 'react';
import appConfig from 'appConfig';

let iconLoader = null;
if (__DEVELOPMENT__) {
  iconLoader = name => function IconSvg() {
    const logo = require(`./svg/${name}.svg`).default;
    const href = __DEVELOPMENT__ ? `#${logo.id}` : logo.url && logo.url.replace(appConfig.assets.cdn, '');
    if (__SERVER__) {
      const regex = /<symbol[^>]*>((.|\s)+?)<\/symbol>/;
      const _svg = logo.content.replace(regex, (match, ...p) => {
        return `${p[0]}`;
      });
      return (
        <svg viewBox={logo.viewBox} className="icon shape-codepen"
             dangerouslySetInnerHTML={{ __html: `${_svg}` }}
        />
      );
    }
    return (
      <svg viewBox={logo.viewBox} className="icon shape-codepen"
           dangerouslySetInnerHTML={{ __html: "<use xlink:href='" + href + "'></use>" }}
      />
    );
  };
} else {
  iconLoader = (name) => {
    return class IconSvg extends React.Component {
      constructor(props) {
        super(props);
        this.state = {
          logo: null
        };

        this.loadIcon(name);
      }

      loadIcon = async (name) => {
        try {
          const svg = await import(`./svg/${name}.svg`);
          this.setState({ logo: svg.default });
        } catch(e) {
          console.log(`SVG load error: ${e} \n stack trace is ${e.stack}`);
        }
      };

      render() {
        const { logo } = this.state;
        if (logo) {
          const href = __DEVELOPMENT__ ?
            `#${logo.id}` :
            logo.url && logo.url.replace(appConfig.assets.cdn, '');

          if (__SERVER__) {
            const regex = /<symbol[^>]*>((.|\s)+?)<\/symbol>/;
            const _svg = logo.content.replace(regex, (match, ...p) => {
              return `${p[0]}`;
            });
            return (
              <svg viewBox={logo.viewBox} className="icon shape-codepen"
                   dangerouslySetInnerHTML={{
                     __html: `${_svg}`
                   }}
              />
            );
          }

          return (
            <svg viewBox={logo.viewBox} className="icon shape-codepen"
                 dangerouslySetInnerHTML={{ __html: "<use xlink:href='" + href + "'></use>" }}
            />
          );
        } else {
          return null;
        }
      }
    };
  };
}

export const  ProfileIcon = iconLoader('ProfileIcon');
