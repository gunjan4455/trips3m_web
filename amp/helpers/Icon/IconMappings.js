const IconsMapping = {
  CabTransfer: 'CabTransfer',
  Drinks: 'DrinkIcon',
  Flight: 'FlightIcon',
  GooglePlay: 'GooglePlay',
  AppStore: 'AppStore',
  Meal: 'Meal',
  TTlogoSecondary: 'TTlogoSecondary',
  HouseBoat: 'HouseBoatIcon',
  SortIcon: 'SortIcon',
  FilterIcon: 'FilterIcon',
  IndianIcon: 'IndianIcon',
  InternationalIcon: 'InternationalIcon',
  IllusAgent: 'IllusAgent',
  IllusMoneySafe: 'IllusMoneySafe',
  IllusSupport: 'IllusSupport',
  IllusDestination: 'IllusDestination',
  LookingIcon: 'LookingIcon',
  CustomPackage: 'CustomPackage',
  ChatIcon: 'ChatIcon',
  ChatIconWhite: 'ChatIconWhite',
  CallUs: 'CallUs',
  CallBackIcon: 'CallBackIcon',
  Confusion: 'Confusion',
  Help: 'Help',
  FBIcon: 'FBIcon',
  GoogleIcon: 'GoogleIcon',
  LockIcon: 'LockIcon',
  TicketsIcon: 'TicketsIcon',
  ArrowNext: 'ArrowNext',
  CalendarIcon: 'CalendarIcon',
  BudgetIcon: 'BudgetIcon',
  CabIconFront: 'CabIconFront',
  NextIcon: 'NextIcon',
  CloseDark: 'CloseDark',
  IllusRequestCallback: 'IllusRequestCallback',
  Illus404: 'Illus404',
  IllusDestinationsOld: 'IllusDestinationsOld',
  IllusDestinations: 'IllusDestinations',
  IllusQualityCheck: 'IllusQualityCheck',
  IllusForgotPassword: 'IllusForgotPassword',
  IllusThanksRequest: 'IllusThanksRequest',
  AllTrips: 'AllTrips',
  Weekendmeal: 'Meal',
  Trekking: 'TrekkingIcon',
  Rafting: 'RaftingIcon',
  Paragliding: 'ParaIcon',
  Rockclimbing: 'RockClimbingIcon',
  Flyingfox: 'FlyingFoxIcon',
  Repelling: 'RepellingIcon',
  Bonfire: 'BonFireIcon',
  Music: 'MusicIcon',
  Dj: 'DJIcon',
  Jeepsafari: 'JeepIcon',
  Swimming: 'SwimmingPoolIcon',
  Photography: 'CameraIcon',
  Deluzeacbus: 'BusIcon',
  Tempotraveler: 'MinIBusIcon',
  Sharedcoach: 'BusIcon',
  Transport: 'MinIBusIcon',
  Luxurycamp: 'CampIcon',
  Eveningsnack: 'EveningSnacksIcon',
  Jumping: 'SkyDvieIcon',
  Prasailing: 'ParaSellingicon',
  Livemusic: 'MusicIcon',
  MealPlanforHotel: 'Meal',
  Meals: 'Meal',
  Hotel: 'HotelIcon',
  Sightseeing: 'SightseeingIcon',
  Cab: 'CabIconFront',
  SharedCoach: 'CabTransfer',
  Complimentarydrinks: 'DrinkIcon',
  Flights: 'FlightIcon',
  Houseboat: 'HouseBoatIcon',
  Cruise: 'CruiseIcon',
  JetSkiTransfer: 'JetSkiIcon',
  Safari: 'SafariIcon',
  Watersports: 'WaterActivities',
  Adventure: 'Adventure',
  Visa: 'VisaIcon',
  FerryTransfer: 'FerriIcon',
  Cake: 'CakeIcon',
  FloralDecoration: 'FloralDecorationIcon',
  Campstay: 'CampIcon',
  Entrytickets: 'TicketsIcon',
  HomeStay: 'HotelIcon',
  JeepSafari: 'JeepIcon',
  TreeHouse: 'TreeHouseIcon',
  Disneyland: 'DisneyLandIcon',
  ReligiousTour: 'ReligiousIcon',
  Transfer: 'CabTransfer',
  Simcard: 'SimCardIcon',
  CloseWhite: 'CloseWhite',
  NotificationBell: 'NotificationBell',
  ShareIcon: 'ShareIcon',
  PackageIcon: 'PackageIcon',
  CompareIcon2: 'CompareIcon2',
  CompareIconWhite: 'CompareIconWhite',
  Uniquerequirements: 'Uniquerequirements',
  Envelope: 'Envelope',
  ThankYouTick: 'ThankYouTick',
  Default: 'DefaultIcon',
  Activities: 'TrekkingIcon',
  AdoutDestinationIcon: 'AdoutDestinationIcon',
  AgentIcon: 'AgentIcon',
  AndroidIcon: 'AndroidIcon',
  AttachmentIcon: 'AttachmentIcon',
  Autumn: 'Autumn',
  BackIcon: 'BackIcon',
  Back2IconWhite: 'Back2IconWhite',
  BackTopArrow: 'BackTopArrow',
  BeachIconGrey: 'BeachIconGrey',
  BudgetIconNew: 'BudgetIconNew',
  BusIllus: 'BusIllus',
  CabIconFrontGrey: 'CabIconFrontGrey',
  CallUsWhite: 'CallUsWhite',
  CallUsWhiteFill: 'CallUsWhiteFill',
  Clock: 'Clock',
  ComparePackage: 'ComparePackage',
  CustomisedPackages: 'CustomisedPackages',
  DefaultIcon: 'DefaultIcon',
  DurationIcon: 'DurationIcon',
  EmiIcon: 'EmiIcon',
  ExclusionIcon: 'ExclusionIcon',
  Ferrytransfer: 'Ferrytransfer',
  FilterIconWhite: 'FilterIconWhite',
  Friends: 'Friends',
  GymIcon: 'GymIcon',
  HalfStar: 'HalfStar',
  HillStation: 'HillStationIcon',
  HillStationIcon: 'HillStationIcon',
  HillStationIconGrey: 'HillStationIconGrey',
  History: 'History',
  HotelEmptyState: 'HotelEmptyState',
  InclusionIcon: 'InclusionIcon',
  IndianIconGrey: 'IndianIconGrey',
  IOS: 'IOS',
  KeyboardActive: 'KeyboardActive',
  KeyboardInactive: 'KeyboardInactive',
  LinkedIn: 'LinkedIn',
  LockWhite: 'LockWhite',
  LoveIcon: 'LoveIcon',
  MailIconWhite: 'MailIconWhite',
  MessageIconNew: 'MessageIconNew',
  MobileIcon: 'MobileIcon',
  Monsoon: 'Monsoon',
  MountainIcon: 'MountainIcon',
  Nature: 'NatureIcon',
  NatureIcon: 'NatureIcon',
  NatureIconGrey: 'NatureIconGrey',
  NotificationIcon: 'NotificationIcon',
  PhoneIconWhite: 'PhoneIconWhite',
  PriceIcon: 'PriceIcon',
  Religious: 'ReligiousIcon',
  FriendsAndGroups: 'ReligiousIcon',
  RestaurantIcon: 'RestaurantIcon',
  RoomServiceIcon: 'RoomServiceIcon',
  SearchBlack: 'SearchBlack',
  Ship: 'Ship',
  StatusCrossIcon: 'StatusCrossIcon',
  StatusSuccessIcon: 'StatusSuccessIcon',
  StepArrow: 'StepArrow',
  Summer: 'Summer',
  TaxiCab: 'TaxiCab',
  Train: 'Train',
  TTIcon: 'TTIcon',
  TwitterGrey: 'TwitterGrey',
  Visa2: 'Visa',
  WaterActivities: 'WaterActivities',
  WaterActivitiesGrey: 'WaterActivitiesGrey',
  WeatherIcon: 'WeatherIcon',
  WifiIcon: 'WifiIcon',
  Winter: 'Winter',
  DiamondIcon: 'DiamondIcon',
  WebsiteIconNew: 'WebsiteIconNew',
  AirConditioningIcon: 'AirConditioningIcon',
  Bar: 'BarIcon',
  Beachaccess: 'BeachAccessIcon',
  FreeBreakfast: 'BreakfastIconNew',
  BusinessCenter: 'BusinessCentreIcon',
  HChildFriendly: 'ChildFriendlyIcon',
  EmailIcon: 'EmailIcon',
  Golfcourse: 'GolfIcon',
  Fitnesscenter: 'GymIconNew',
  HotTub: 'HotTubIcon',
  Kitcheninsomerooms: 'KitchenIcon',
  LaundryService: 'LaundryIcon',
  LocationIcon: 'LocationIcon',
  Freeparking: 'ParkingIcon',
  PetFriendly: 'PetFriendlyIcon',
  PhoneIcon: 'PhoneIcon',
  OutdoorPool: 'PoolIcon',
  Restaurant: 'RestaurantIconNew',
  RoomService: 'RoomServiceIconNew',
  AirportShuttle: 'ShuttleIcon',
  Smokefree: 'SmokeFreeIcon',
  Spa: 'SpaIcon',
  WalletIcon: 'WalletIcon',
  TTRecommendedIcon: 'TTRecommendedIcon',
  BulbIcon: 'BulbIcon',
  WhatsAppIcon: 'WhatsAppIcon',
  TajMahalIcon: 'TajMahalIcon',
  AdvantureIconNew: 'AdvantureIconNew',
  PlusInclusionIcon: 'PlusInclusionIcon',
  IntentExplore: 'IntentExplore',
  WhiteShareIcon: 'ShareWhite',
  LandlineIcon: 'LandlineIcon',
  DownloadVoucherIcon: 'DownloadVoucherIcon',
  EditPreferencesIcon: 'EditPreferencesIcon',
  HeadphoneIcon: 'HeadphoneIcon',
  HeadphoneIconWhite: 'HeadphoneIconWhite',
  ShareBlack: 'ShareBlack',
  NotificationIconWhite: 'NotificationIconWhite',
  DownloadBlack: 'DownloadBlack',
  DownloadBlue: 'DownloadBlue',
  IllusProgressB: 'IllusProgressB',
  GmailIcon: 'GmailIcon',
  DateDecided: 'DateDecided',
  DateNotDecided: 'DateNotDecided',
  BudgetLow: 'BudgetLow',
  BudgetMedium: 'BudgetMedium',
  BudgetHigh: 'BudgetHigh',
  Travelers: 'Travelers',
  Contact: 'Contact',
  Destination: 'Destination',
  LeftRedArrow: 'LeftRedArrow',
  RightWhiteArrow: 'RightWhiteArrow',
  DisabledArrowLeft: 'DisabledArrowLeft',
  HotelFilled: 'HotelFilled',
  FlightFilled: 'FlightFilled',
  CabsFilled: 'CabsFilled',
  BudgetMin: 'BudgetMin',
  Tick: 'Tick',
  PartPay: 'PartPay',

  // QDP new Icons
  RequestIcon: 'RequestIcon',
  PreferencesIcon: 'PreferencesIcon',
  InclusionsNewIcon: 'InclusionsNewIcon',
  ExclusionsNewIcon: 'ExclusionsNewIcon',
  TnCIcon: 'TnCIcon',
  InfoBlackIcon: 'InfoBlackIcon',
  ItineraryIcon: 'ItineraryIcon',
  TravelInfoIcon: 'TravelInfoIcon',
  SuccessIllustration: 'SuccessIllustration',
  BriefcaseWhiteIcon: 'BriefcaseWhiteIcon',
  FamilyWhiteIcon: 'FamilyWhiteIcon',
  HoneymoonWhiteIcon: 'HoneymoonWhiteIcon',
  TriangleIcon: 'TriangleIcon',
  ProgressIcon: 'ProgressIcon',
  ProcessedIcon: 'ProcessedIcon',
  EnvelopeIcon: 'EnvelopeIcon',
  PlayIcon: 'PlayIcon',
  CompareIcon: 'CompareIcon',
  ChatIconGrey: 'ChatIconGrey',
  LuxuryPackageIcon: 'LuxuryPackageIcon',
  BlackShareIcon: 'Black-share-icon',
  QuotesReceivedIcon: 'QuotesReceivedIcon',
  NoQuotesReceivedIcon: 'NoQuotesReceivedIcon',
  OutlineStarIcon: 'OutlineStarIcon',
  FillStarIcon: 'FillStarIcon',
  QualityQuotesIcon: 'QualityQuotesIcon',
  RightGreyArrow: 'RightGreyArrow',
  CloseIconWhite: 'CloseIconWhite',
  CrossRed: 'CrossRed',
  GreenTickInclusion: 'GreenTickInclusion',
  ExclusionsRed: 'ExclusionsRed',
  PriceRupee: 'PriceRupee',
  WaveIcon: 'WaveIcon',
  MktAgents: 'MktAgents',
  MktDestinations: 'MktDestinations',
  MktVerified: 'MktVerified',
  MktQualityControl: 'MktQualityControl',
  MktSupport: 'MktSupport',
  MktSelectPackage: 'MktSelectPackage',
  MktQuotes: 'MktQuotes',
  MktTravelers: 'MktTravelers',
  ProfileIcon2: 'ProfileIcon2',


  //amp icons
  IllusTraveler: 'IllusTraveler',
  PdpQuotes: 'PdpQuotes',
  PdpSelectPackage: 'PdpSelectPackage',
  DownArrow: 'DownArrow',
  StarIcon: 'StarIcon',
  StarIconEmpty: 'StarIconEmpty',
  Back2Icon: 'Back2Icon',
  InfoIcon: 'InfoIcon',
  SearchIcon: 'SearchIcon',
  TTlogo: 'TTlogo',
  ShareWhite: 'ShareWhite',
  ProfileIcon: 'ProfileIcon',
  HomeIcon: 'HomeIcon',
  FamilyIcon: 'FamilyIcon',
  WorkIcon: 'WorkIcon',
  OffersIcon: 'OffersIcon',
  BlogIcon: 'BlogIcon',
  TestimonailsIcon: 'TestimonailsIcon',
  FAQIcon: 'FAQIcon',
  ContactIcon: 'ContactIcon',
  VactionIcon: 'VactionIcon',
  HoneymoonPackages: 'HoneymoonPackages',
  SeasonalIcon: 'SeasonalIcon',
  InternationalIconBlack: 'InternationalIconBlack',
  LocationMarkerIcon: 'LocationMarkerIcon',
  FBIconWhite: 'FBIconWhite',
  Twitter: 'Twitter',
  PinterestIcon: 'PinterestIcon',
  GoogleIconWhite: 'GoogleIconWhite',
  Instagram: 'Instagram',
  BlueDownArrow: 'BlueDownArrow',
  GreenTick: 'GreenTick',



};

export default IconsMapping;

// const ampIconMappings = {
//   IllusTraveler: 'IllusTraveler',
//   PdpQuotes: 'PdpQuotes',
//   PdpSelectPackage: 'PdpSelectPackage',
//   DownArrow: 'DownArrow',
//   StarIcon: 'StarIcon',
//   StarIconEmpty: 'StarIconEmpty',
//   Back2Icon: 'Back2Icon',
//   InfoIcon: 'InfoIcon',
//   SearchIcon: 'SearchIcon',
//   TTlogo: 'TTlogo',
//   ShareWhite: 'ShareWhite',
//   ProfileIcon: 'ProfileIcon',
//   HomeIcon: 'HomeIcon',
//   FamilyIcon: 'FamilyIcon',
//   WorkIcon: 'WorkIcon',
//   OffersIcon: 'OffersIcon',
//   BlogIcon: 'BlogIcon',
//   TestimonailsIcon: 'TestimonailsIcon',
//   FAQIcon: 'FAQIcon',
//   ContactIcon: 'ContactIcon',
//   VactionIcon: 'VactionIcon',
//   HoneymoonPackages: 'HoneymoonPackages',
//   SeasonalIcon: 'SeasonalIcon',
//   InternationalIconBlack: 'InternationalIconBlack',
//   LocationMarkerIcon: 'LocationMarkerIcon',
//   FBIconWhite: 'FBIconWhite',
//   Twitter: 'Twitter',
//   PinterestIcon: 'PinterestIcon',
//   GoogleIconWhite: 'GoogleIconWhite',
//   Instagram: 'Instagram',
//   BlueDownArrow: 'BlueDownArrow',
//   GreenTick: 'GreenTick',
// };
