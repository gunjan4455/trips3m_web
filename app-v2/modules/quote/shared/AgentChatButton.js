import React from 'react';
import PropTypes from 'prop-types';

import { DefaultIcon } from 'helpers/Icon/Icon';
import Modal from 'modules/shared/GenericModal';
import Chat from 'modules/chat';

class AgentChatButton extends React.Component {
  static propTypes = {
    name: PropTypes.string,
    showIcon: PropTypes.bool,
    handleChatClick: PropTypes.func,
    textClassName: PropTypes.string,
    customButton: PropTypes.bool
  };

  static defaultProps = {
    name: 'Chat',
    showIcon: false,
    handleChatClick: () => {},
    textClassName: 'f14 fw7 sfc1',
    customButton: false
  };

  constructor(props) {
    super(props);
  }

  render() {
    const { textClassName } = this.props;
    const ChatContainer = Chat.container;

    return (
      <div>
        {
          this.props.showIcon
            ? <span className="input-24 mr8"><DefaultIcon /></span>
            : null
        }
        <Modal
          trigger={
            <button className={`${textClassName} at_agent_chat_button`} onClick={this.props.handleChatClick}>{this.props.name}</button>
          }
          fullView={true}
          hasCustomClose={true}
          isWhiteIcon={true}
        >
          <ChatContainer />
        </Modal>
      </div>
    );
  }
}

export default AgentChatButton;

