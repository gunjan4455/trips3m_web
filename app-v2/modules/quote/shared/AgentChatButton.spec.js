import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import AgentChatButton from './AgentChatButton';
import { DefaultIcon } from 'helpers/Icon/Icon';

configure({ adapter: new Adapter() });

const shallowWrappperAgentChatButton = (props) => {
  return shallow(<AgentChatButton {...props} />);
};

const DEFAULT_PROPS = {
  name: 'Chat',
  showIcon: false,
  handleChatClick: jest.fn,
  textClassName: 'sfc1'
};

describe('AgentChatButton: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperAgentChatButton(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show default icon when showIcon props is given', () => {
    props.showIcon = true;
    const wrapper = shallowWrappperAgentChatButton(props);
    expect(wrapper).toMatchSnapshot();
  });

});
