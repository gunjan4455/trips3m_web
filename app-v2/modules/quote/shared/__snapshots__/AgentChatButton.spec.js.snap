// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentChatButton: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AgentChatButton
    customButton={false}
    handleChatClick={[Function]}
    name="Chat"
    showIcon={false}
    textClassName="sfc1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <ModalComponent
          customClass=""
          customCloseTrigger={false}
          disabled={false}
          fullView={true}
          hasCustomClose={true}
          hasNoClose={false}
          isTriggerFullView={false}
          isWhiteIcon={true}
          onAfterModalOpen={[Function]}
          onRequestCloseCustom={[Function]}
          trigger={
            <button
              className="sfc1 at_agent_chat_button"
              onClick={[Function]}
            >
              Chat
            </button>
          }
        >
          <withRouter(Connect(Chat)) />
        </ModalComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withRouter(Connect(Chat)) />,
          "customClass": "",
          "customCloseTrigger": false,
          "disabled": false,
          "fullView": true,
          "hasCustomClose": true,
          "hasNoClose": false,
          "isTriggerFullView": false,
          "isWhiteIcon": true,
          "onAfterModalOpen": [Function],
          "onRequestCloseCustom": [Function],
          "trigger": <button
            className="sfc1 at_agent_chat_button"
            onClick={[Function]}
          >
            Chat
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <ModalComponent
            customClass=""
            customCloseTrigger={false}
            disabled={false}
            fullView={true}
            hasCustomClose={true}
            hasNoClose={false}
            isTriggerFullView={false}
            isWhiteIcon={true}
            onAfterModalOpen={[Function]}
            onRequestCloseCustom={[Function]}
            trigger={
              <button
                className="sfc1 at_agent_chat_button"
                onClick={[Function]}
              >
                Chat
              </button>
            }
          >
            <withRouter(Connect(Chat)) />
          </ModalComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(Connect(Chat)) />,
            "customClass": "",
            "customCloseTrigger": false,
            "disabled": false,
            "fullView": true,
            "hasCustomClose": true,
            "hasNoClose": false,
            "isTriggerFullView": false,
            "isWhiteIcon": true,
            "onAfterModalOpen": [Function],
            "onRequestCloseCustom": [Function],
            "trigger": <button
              className="sfc1 at_agent_chat_button"
              onClick={[Function]}
            >
              Chat
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`AgentChatButton: Render Correctly  should show default icon when showIcon props is given 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AgentChatButton
    customButton={false}
    handleChatClick={[Function]}
    name="Chat"
    showIcon={true}
    textClassName="sfc1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="input-24 mr8"
        >
          <IconSvg />
        </span>,
        <ModalComponent
          customClass=""
          customCloseTrigger={false}
          disabled={false}
          fullView={true}
          hasCustomClose={true}
          hasNoClose={false}
          isTriggerFullView={false}
          isWhiteIcon={true}
          onAfterModalOpen={[Function]}
          onRequestCloseCustom={[Function]}
          trigger={
            <button
              className="sfc1 at_agent_chat_button"
              onClick={[Function]}
            >
              Chat
            </button>
          }
        >
          <withRouter(Connect(Chat)) />
        </ModalComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "input-24 mr8",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withRouter(Connect(Chat)) />,
          "customClass": "",
          "customCloseTrigger": false,
          "disabled": false,
          "fullView": true,
          "hasCustomClose": true,
          "hasNoClose": false,
          "isTriggerFullView": false,
          "isWhiteIcon": true,
          "onAfterModalOpen": [Function],
          "onRequestCloseCustom": [Function],
          "trigger": <button
            className="sfc1 at_agent_chat_button"
            onClick={[Function]}
          >
            Chat
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="input-24 mr8"
          >
            <IconSvg />
          </span>,
          <ModalComponent
            customClass=""
            customCloseTrigger={false}
            disabled={false}
            fullView={true}
            hasCustomClose={true}
            hasNoClose={false}
            isTriggerFullView={false}
            isWhiteIcon={true}
            onAfterModalOpen={[Function]}
            onRequestCloseCustom={[Function]}
            trigger={
              <button
                className="sfc1 at_agent_chat_button"
                onClick={[Function]}
              >
                Chat
              </button>
            }
          >
            <withRouter(Connect(Chat)) />
          </ModalComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "input-24 mr8",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(Connect(Chat)) />,
            "customClass": "",
            "customCloseTrigger": false,
            "disabled": false,
            "fullView": true,
            "hasCustomClose": true,
            "hasNoClose": false,
            "isTriggerFullView": false,
            "isWhiteIcon": true,
            "onAfterModalOpen": [Function],
            "onRequestCloseCustom": [Function],
            "trigger": <button
              className="sfc1 at_agent_chat_button"
              onClick={[Function]}
            >
              Chat
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
